src/_test_/Detail.test.js

```diff
@@ -0,0 +1,27 @@
+import { render, screen } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from '../redux/configureStore';
+import Details from '../utils/Detail';
+import '@testing-library/jest-dom/extend-expect';
+
+test('should render Details', () => {
+  const tree = render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Details />
+      </BrowserRouter>
+    </Provider>,
+  );
+  expect(tree).toMatchSnapshot();
+});
+test('should render Rank', () => {
+  render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Details />
+      </BrowserRouter>
+    </Provider>,
+  );
+  expect(screen.getByTestId('Details-1')).toHaveTextContent('Rank');
+});
```

Comment:
- [ ] **[OPTIONAL]:** I suggest that you take a look at the [article on writing tests that interact with redux](https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function), the official documentation suggests not to use the same instance of the store for all the tests because in some cases it can cause unexpected dependency between them. Instead, they provide a custom render function that creates a store instance reusing your reducers, and solving this problem! 🤓 🔧